--- a/kernel/supercalls.h
+++ b/kernel/supercalls.h
@@ -9,13 +9,14 @@
 #define KSU_INSTALL_MAGIC1 0xDEADBEEF
 #define KSU_INSTALL_MAGIC2 0xCAFEBABE
 
 // Toolkit extensions
 #define CHANGE_MANAGER_UID 10006
-#define KSU_UMOUNT_GETSIZE 107
-#define KSU_UMOUNT_GETLIST 108
-#define GET_SULOG_DUMP_V2 10010
+#define GET_SULOG_DUMP 10009        // get sulog dump, max, last 100 escalations
+#define GET_SULOG_DUMP_V2 10010     // get sulog dump, max, last 250 escalations
+#define KSU_UMOUNT_GETSIZE 107   // get list size // shit is u8 we cant fit 10k+ on it
+#define KSU_UMOUNT_GETLIST 108   // get list
 
 // Command structures for ioctl
 
 struct ksu_become_daemon_cmd {
 	__u8 token[65]; // Input: daemon token (null-terminated)
